packages/as-proto-gen/src/file-context.ts

```diff
@@ -44,8 +51,57 @@ export class FileContext {
     return [uniqueImportName, ...importNamespace].join(".");
   }
 
-  registerDefinition(definitionNamePath: string): string {
-    const [definitionName] = definitionNamePath.split(".");
+  getRelativeImportPath(
```

Comment:
We can't because the getRelativeImportPath also truncate path suffix to keep "namespacing" for imported object.

For example, if `google.protobuf.Timestamp` is imported from `cosmos/authz/v1beta1/authz.ts` it will be imported as `../../../google`